### ISSUE TITLE
Compatibility with org-modern

### ISSUE THREAD

**Issue by JulienMalka** [alphapapa/org-ql#477]
Hello, 
I know that this has been discussed before, but it looks like `org-ql` is not compatible with the `org-modern` package. For some reason, `org-agenda-custom-commands` using `org-ql-block` doesn't have the right styling for TODO keywords, although the styling for the tags is correct.
I join here a screenshot of the problem and a reproducer. 

![1733342582-wayshot](https://github.com/user-attachments/assets/e387cdfe-2764-4d3a-965d-9cf1aa275ce3)

```
(require 'package)
(require 'use-package)

;; Add MELPA for packages
(add-to-list 'package-archives '("melpa" . "https://melpa.org/packages/") t)

;; Make sure we install updated org from the repositories
(assq-delete-all 'org package--builtins)
(assq-delete-all 'org package--builtin-versions)


(use-package org-ql 
 :ensure t)

(use-package org
  :custom
  ;; Tell Org where to find all my stuff
  (org-directory "~/dev/org")
  (org-agenda-files '("~/dev/todos/test.org"))

  ;; My main custom agenda view
  (org-agenda-custom-commands
   '(("c" "Primary agenda view"
      (
       (org-ql-block '(and
		(todo "TODO")
		(tags "inbox")))

        (tags-todo "inbox"
        	((org-agenda-prefix-format "  %?-12t% s")
                (org-agenda-overriding-header "Inbox"))))))))


(use-package org-modern
  ;;:hook ((org-mode . org-modern-mode)
  ;;       (org-agenda-finalize . org-modern-agenda))
  :ensure t
  :config
  (global-org-modern-mode)
  :custom
  (org-modern-star 'replace)
  ;; Set manually to match leuven-theme
  (org-modern-todo-faces
   `(("TODO" :background "#DBEDFF" :foreground "#333333")
     ("NEXT" :background "#FFC8C8" :foreground "#333333")
     ("HOLD" :background "#F6FECD" :foreground "#333333")
     ("DONE" :background "#D5F1CF" :foreground "#333333"))))
     ```

**Comment by JulienMalka** [comment]
Not knowing exactly who is in a better position to address this, I've opened a sister issue here https://github.com/minad/org-modern/issues/245

**Comment by alphapapa** [comment]
The issue is probably one of having the appropriate text property on the line.  You can see in https://github.com/alphapapa/org-ql/issues/455 where I recently removed text properties from the to-do keyword because it was sometimes picking up other properties from the buffer applied by other modes, which caused incorrect appearance in org-ql-view buffers.  So basically, what needs to be done is:

1.  Identify the text property and values used by org-modern to recognize the to-do keyword.  This can usually be done with `C-u C-x =` on the lines in the agenda, comparing the ones that appear as desired and the ones that don't.
2. Add code in `org-ql-view--format-element` to set that property and value accordingly.

It should be a relatively simple fix.  If you have time to investigate step 1, that would help.  My time to work on Emacs stuff lately has been limited.

**Comment by JulienMalka** [comment]
Yes I can confirm that those two properties are indeed not present with `org-ql`.
